flink/src/main/java/org/apache/iceberg/flink/IcebergTableSource.java

```diff
@@ -33,33 +37,37 @@
 import org.apache.flink.table.types.DataType;
 import org.apache.flink.table.utils.TableConnectorUtils;
 import org.apache.iceberg.flink.source.FlinkSource;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
 
 /**
  * Flink Iceberg table source.
- * TODO: Implement {@link FilterableTableSource}
  */
 public class IcebergTableSource
-    implements StreamTableSource<RowData>, ProjectableTableSource<RowData>, LimitableTableSource<RowData> {
+    implements StreamTableSource<RowData>, ProjectableTableSource<RowData>, FilterableTableSource<RowData>,
+    LimitableTableSource<RowData> {
 
   private final TableLoader loader;
   private final TableSchema schema;
   private final Map<String, String> properties;
   private final int[] projectedFields;
   private final boolean isLimitPushDown;
   private final long limit;
+  private final List<org.apache.iceberg.expressions.Expression> filters;
 
   public IcebergTableSource(TableLoader loader, TableSchema schema, Map<String, String> properties) {
-    this(loader, schema, properties, null, false, -1);
+    this(loader, schema, properties, null, false, -1, null);
```

Comment:
It's better to use `ImmutableList.of()`  as the default `filters` because of the comment [here](https://github.com/apache/iceberg/pull/1893/files#r544779804), though there's large probability that when `isFilterPushedDown()` returns true the filters should always be non-nullable.